exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsConfigHelper.java

```diff
@@ -0,0 +1,126 @@
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+
+/**
+ * Utility class to help with management of TLS related components. This class is ultimately
+ * responsible for enabling TLS with the OkHttpClientBuilder by setting the sslSocketFactory. This
+ * class is only intended for internal OpenTelemetry exporter usage and should not be used by
+ * end-users.
+ */
+public class TlsConfigHelper {
+
+  private static final Logger logger = Logger.getLogger(TlsConfigHelper.class.getName());
+
+  @Nullable private X509KeyManager keyManager;
+  @Nullable private X509TrustManager trustManager;
+  @Nullable private SSLSocketFactory sslSocketFactory;
+
+  public TlsConfigHelper setTrustManager(X509TrustManager trustManager) {
+    this.trustManager = trustManager;
+    return this;
+  }
+
+  /**
+   * Creates a new X509TrustManager from the given cert content.
+   *
+   * @param trustedCertsPem Certificate in PEM format.
+   * @return this
+   */
+  public TlsConfigHelper configureTrustManager(byte[] trustedCertsPem) {
+    try {
+      this.trustManager = TlsUtil.trustManager(trustedCertsPem);
+    } catch (SSLException e) {
+      throw new IllegalStateException(
+          "Error creating trust manager for OTLP HTTP connection with provided certs. Are they valid X.509 in PEM format?",
+          e);
+    }
+    return this;
+  }
+
+  /**
+   * Creates a new X509KeyManager from the given private key and certificate, both in PEM format.
+   *
+   * @param privateKeyPem Private key content in PEM format.
+   * @param certificatePem Certificate content in PEM format.
+   * @return this
+   */
+  public TlsConfigHelper configureKeyManager(byte[] privateKeyPem, byte[] certificatePem) {
+    try {
+      if (keyManager != null) {
+        logger.warning(
+            "Previous X509 Key manager is being replaced. This is probably an error and should only be set once.");
+      }
+      keyManager = TlsUtil.keyManager(privateKeyPem, certificatePem);
+      return this;
+    } catch (SSLException e) {
+      throw new IllegalStateException(
+          "Error creating key manager for TLS configuration of OTLP HTTP connection with provided certs. Are they valid X.509 in PEM format?",
+          e);
+    }
+  }
+
+  /**
+   * Assigns the X509KeyManager.
+   *
+   * @return this
+   */
+  public TlsConfigHelper setKeyManager(X509KeyManager keyManager) {
+    if (this.keyManager != null) {
+      logger.warning(
+          "Previous X509 Key manager is being replaced. This is probably an error and should only be set once.");
+    }
+    this.keyManager = keyManager;
+    return this;
+  }
+
+  public TlsConfigHelper setSslSocketFactory(SSLSocketFactory sslSocketFactory) {
+    this.sslSocketFactory = sslSocketFactory;
+    return this;
+  }
+
+  /** Configures TLS for the provided OkHttp client builder by setting the SSL Socket Factory. */
+  public void configure(OkHttpClient.Builder clientBuilder) {
```

Comment:
How about having this accept a `Consumer<SSLSocketFactory>` so this isn't coupled to okhttp?

Reply:
Yeah totally....I started down this path when I realized there are like 2 or 3 other implementations doing exactly the same thing.

Reply:
So please have a look at how it ended up. Unfortunately, the managed channel stuff inside GRPC uses a different approach and I had to create two different callback interfaces. I think it worked out ok. I also wanted to be able to handle `SSLException` in the helper, which required the checked exception to be on the interface method.